### DISCUSSION TITLE
How to handle money: int in db, float in pydantic

### DISCUSSION THREAD

**Discussion by ArturPrzybysz** [pydantic/pydantic#7664] [Question] [answered]
Hi, I am trying to accomplish elegantly the following workflow; in my db I store prices in integers (cents), to avoid any floating point errors. However, my frontend expects a float (dollars) in a request.
Obviously, I could ask fronted to divide the integer on their side, however this is not an option here.
Another option would be to create a different, non-pydantic object from the pydantic one, so that the division by 100 happens there, and then again map it to pydantic. This however seems wasteful.

How can I achieve this using only the pydantic class?
```
# my shortened db definition
class Item(Base):
    id: Mapped[str] = mapped_column(
        UUID(as_uuid=False), primary_key=True, server_default=text("gen_random_uuid()"), unique=True
    )
    cost_per_hour: Mapped[int] = mapped_column(Integer)
```
```
# my pydantic class
class Item(BaseModel):
    id: str
    cost_per_hour: float # divides model's cost_per_hour by 100, to obtain price in dollars.
```

**Comment by dimi-colossal** [comment, marked as answer]
I believe you need a [computed field](https://docs.pydantic.dev/2.0/usage/computed_fields/)

**Reply by ArturPrzybysz** [reply]
That's true! This is what I ended up using.
